net/blockwise/blockwise.go

```diff
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	udpPool "github.com/plgd-dev/go-coap/v2/udp/message/pool"
```

Comment:
This is ugly. It will be better when we move github.com/plgd-dev/go-coap/v2/udp/message/type.go to ithub.com/plgd-dev/go-coap/v2/message/types/type.go and extends the Message interface about SetType and Type functions (for tcp it do will be do nothing) and for UDP it will be proper used.
What do you think ?

Reply:
Was thinking the same. Maybe even a new interface just for the "Type" thing.
I just don't wanted to change the generic message package. I will update the PR to avoid the dependency to the UDP package here.

Reply:
I would move it to `github.com/plgd-dev/go-coap/v2/message/type.go`

Reply:
Any specific reasons for a separate `types` package? I would keep the number of packages lower and just put it to `.../go-coap/v2/message/type.go`

I can see that `codes` have their own package. Whats the motivations?

Reply:
`types` directly doesn't belongs to common message (`go-coap/v2/message`). Maybe you can leave it in `github.com/plgd-dev/go-coap/v2/udp/message` and import that package.

Sure I m ok with simple interface `Type`

Reply:
It's a question of what the expected scope of `go-coap/v2/message` is. 

I moved it back, but are using the interface now.

BTW thanks for invinting me to the project. I would still leave the actual merging to you.

Reply:
I m happy from with your contribution 👍